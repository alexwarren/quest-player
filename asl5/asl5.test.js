'use strict';

const output = [];

window.addTextAndScroll = (text) => {
    output.push(text);
};

window.uiShow = () => {};
window.uiHide = () => {};
window.scrollToEnd = () => {};

const asl5 = require('./asl5');
const fs = require('fs');

test('loads test.aslx', () => {
    const data = fs.readFileSync('examples/test.aslx', 'utf-8');
    asl5.load(data);
    asl5.begin();
    asl5.sendCommand('hello');
    asl5.sendCommand('fn');
    asl5.sendCommand('invoke');
    asl5.sendCommand('for');
    asl5.sendCommand('foreach');
    asl5.sendCommand('hasstring');
    asl5.sendCommand('a');
    asl5.sendCommand('attrs');
    asl5.sendCommand('input');
    asl5.sendCommand('test some input for GetInput...');
    asl5.sendCommand('sw');
    expect(output).toMatchSnapshot();
});