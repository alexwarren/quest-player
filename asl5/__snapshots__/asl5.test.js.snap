// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads test.aslx 1`] = `
Array [
  "Hello world 123, 456<br/>",
  "Welcome<br/>",
  "Test attributes...<br/>",
  "game.testattr = Test attribute value<br/>",
  "game.testattr2 = Test another attribute value<br/>",
  "game.inheritedattr = An inherited attribute value<br/>",
  "game.overrideattr = This is the overridden attribute value<br/>",
  "game.overrideattr2 = This is another overridden attribute value<br/>",
  "Greetings!<br/>",
  "call fn<br/>",
  "start fn<br/>",
  "Result 10<br/>",
  "arg: 2<br/>",
  "running script...<br/>",
  "This was passed as a script argument.<br/>",
  "done invoke<br/>",
  "1<br/>",
  "2<br/>",
  "3<br/>",
  "4<br/>",
  "5<br/>",
  "6<br/>",
  "7<br/>",
  "8<br/>",
  "9<br/>",
  "10<br/>",
  "running foreach...<br/>",
  "northwest<br/>",
  "north<br/>",
  "northeast<br/>",
  "west<br/>",
  "east<br/>",
  "southwest<br/>",
  "south<br/>",
  "southeast<br/>",
  "up<br/>",
  "down<br/>",
  "in<br/>",
  "out<br/>",
  "done<br/>",
  "testattr true<br/>",
  "madeupattr false<br/>",
  "compassdirections false<br/>",
  "Get values...<br/>",
  "testattr Test attribute value<br/>",
  "madeupattr null<br/>",
  "compassdirections null<br/>",
  "compassdirections as boolean false<br/>",
  "abcdef<br/>",
  "name,elementtype,type,gamename,testattr,testattr2,overrideattr,compassdirections,overrideattr2,inheritedattr<br/>",
  "Getting input...<br/>",
  "test some input for GetInput...<br/>",
  "Done<br/>",
  "Something else<br/>",
  "Done switch<br/>",
]
`;
